apps/docs/features/docs/Troubleshooting.utils.ts

```diff
@@ -27,7 +27,9 @@ export interface ITroubleshootingEntry {
 export const getArticleSlug = getArticleSlugInternal
 
 async function getAllTroubleshootingEntriesTyped() {
-  const result = (await getAllTroubleshootingEntriesInternal()) as ITroubleshootingEntry[]
+  const result = (
+    IS_PLATFORM ? await getAllTroubleshootingEntriesInternal() : []
+  ) as ITroubleshootingEntry[]
```

Comment:
Just a casting nit: try to avoid casting unless you have no other option, otherwise it defeats the purpose of type safety. I'm wondering if `getAllTroubleshootingEntriesInternal()` can explicitly return `ITroubleshootingEntry[]`, then you can just do:

 ```suggestion
  const result: ITroubleshootingEntry[] = (
    IS_PLATFORM ? await getAllTroubleshootingEntriesInternal() : []
  )
```

Reply:
This problem occurs because we don't have `strictNullChecks` on (too many existing violations in the code base, need to find time to sit down and grind through all of them 😬), so Zod doesn't return the right required types even though the schema validation enforces it.

I either have to cast or deal with fake optional fields all the way down, so I figure casting is less annoying. Root fix is just to grit my teeth and get `strictNullChecks` done, I'll make a separate task for that!

Reply:
Pesky legacy code! Makes sense 😄